File: test/resources/repos/csharp/test_repo/MultipleInterfaces.cs
using System;

namespace TestRepo
{
    public class MultipleInterfaces : IReadable, IWritable, IProcessable
    {
        private string data;

        public MultipleInterfaces(string data)
        {
            this.data = data;
        }

        public string Read()
        {
            return data;
        }

        public void Write(string data)
        {
            this.data = data;
        }

        public void Process()
        {
            Console.WriteLine($"Processing data: {data}");
        }

        public string GetType()
        {
            return "MultipleInterfaces";
        }
    }
}

File: test/resources/repos/typescript/test_repo/MultipleInterfaces.ts
import { Readable, Writable, Processable } from './BaseClass';

export class MultipleInterfaces implements Readable, Writable, Processable {
    private data: string;

    constructor(data: string) {
        this.data = data;
    }

    read(): string {
        return this.data;
    }

    write(data: string): void {
        this.data = data;
    }

    process(): void {
        console.log(`Processing data: ${this.data}`);
    }

    getType(): string {
        return 'MultipleInterfaces';
    }
}